### ISSUE TITLE
PEGylated lipid bilayer containing MARTINI 3 cholesterol. 

### ISSUE THREAD

**Issue by ohyh** [marrink-lab/polyply_1.0#332]
polyply version 1.4.0
polyply gen_coords -p system_10.top -o 2k_010.gro -c memb.gro -res PEO OHter -name test -box 15 15 20 -split DPPE:HEAD-NH3:TAIL-PO4,GL1,GL2,C1A,C2A,C3A,C4A,C1B,C2B,C3B,C4B

**Expected behavior**
Hi Polyply team,

I was trying to generate PEG chains on the membrane including cholesterols. I got configuration with two problems - 1) PEG chains imbedded inside of the membrane, and 2) floating cholesterol out of the membrane. I assume MARTINI3 cholesterol is recently release, and Polyply hasen't been updated? Please, take a look into the files that I uploaded. 

**Additional Files**
If applicable, add all files needed in the reproduction.
[tmp.zip](https://github.com/marrink-lab/polyply_1.0/files/11883016/tmp.zip)

**Comment by fgrunewald** [comment]
@ohyh thanks for raising this issue. The reason for the free floating cholesterols is easy to identify: Your topology specifies 216 Cholesterol molecules but in the memb.gro file there are only 192 Cholesterol molecules present. Thus polyply constructs the missing molecules and will just randomly place them. If you fix the Cholesterol in your topology file this should not happen anymore. Be aware to reduce the number of molecules for the correct leaflet. 

The reason for the PEG going inside the membrane is likely due to the fact that your membrane is not equilibrated. There is simply too much space between the lipids. To fix this you can simply provide a build file, which specifies that there should be no PEO inside the membrane. In your case the following file does the job:
```
[ molecule ]
PGL2k 0 36
[ rectangle ]
; resname start stop in/out x y z a b c
; for syntax see wiki
PEO        2     47 out  10.0 10.0 10.0 10.0 10.0 1.5
```
The syntax for the build file is explained [here](https://github.com/marrink-lab/polyply_1.0/wiki/Syntax:-build-file) in the wiki. The polyply command then becomes:
```
polyply gen_coords -p system_10.top -o 2k_010.gro -c memb.gro -res PEO OHter -name test -box 15 15 20 -split DPPE:HEAD-NH3:TAIL-PO4,GL1,GL2,C1A,C2A,C3A,C4A,C1B,C2B,C3B,C4B -b build_file.bld
```
Also please keep in mind that PEG currently doesn't work well with Martini3 lipids. The current workaround is to scale PEG water interactions at the expensive of making PEG more hydrophilic of course.

**Comment by ohyh** [comment]
Hi, thanks for the quick response. I fixed the PEG chain internalization issue by using build file. Thank you.

However, I still got the free cholesterol problem. I double-check that my memb.gro file contains 216 cholesterols, and Polyply adds 27 more cholesterol in the system. Following is the way that I checked my configuration. 

(py3) [ Tue Jun 27 13:18:10 Macbook Pro ] grep -c CHOL memb.gro 
1728
(py3) [ Tue Jun 27 13:18:17 Macbook Pro ] grep -c CHOL 2k_010.gro 
1944

There are 8 particles per cholesterol, so  216 cholesterols in memb.gro, and 243 cholesterols in the output file. I just remove the floating cholesterol and solvate my system. But I'm curious where this weird behavior comes from. 

Again, I appreciate your timely support.
